packages/jaeger-ui/src/utils/tracking/ga.tsx

```diff
@@ -24,6 +23,40 @@ import { logTrackingCalls } from './utils';
 import { getAppEnvironment, shouldDebugGoogleAnalytics } from '../constants';
 import parseQuery from '../parseQuery';
 
+// Modify the `window` object to have an additional attribute `dataLayer`
+// This is required by the gtag.js script to work
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export interface WindowWithGATracking extends Window {
+  dataLayer: (string | object)[][] | undefined;
+}
+
+declare let window: WindowWithGATracking;
+
+// Function to add a new event to the Google Analytics dataLayer
+const gtag = (...args: (string | object)[]) => {
+  if (window !== undefined)
+    if (window.dataLayer !== undefined) {
```

Comment:
Why are you relying on precisely `undefined` here? What about this:
```suggestion
  if (window)
    if (window.dataLayer && isArray(window.dataLayer)) {
```

Reply:
Sign catch!